internal/vendors/service.go

```diff
@@ -43,6 +45,26 @@ func (v *VendorService) GetByProduct(ctx context.Context, product string) ([]Ven
 	return v.vendorDBAccessor.GetByProductDescription(ctx, productDescription)
 }
 
+func (v *VendorService) Put(ctx context.Context, vendor Vendor) (*Vendor, error) {
+	currentVendor, err := v.vendorDBAccessor.GetById(ctx, vendor.ID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	newVendor := Vendor(*currentVendor)
+	newVendor.Name = vendor.Name
+	newVendor.Description = vendor.Description
+	newVendor.BpID = vendor.BpID
+	newVendor.BpName = vendor.BpName
+	newVendor.Rating = vendor.Rating
+	newVendor.AreaGroupID = vendor.AreaGroupID
+	newVendor.AreaGroupName = vendor.AreaGroupName
+	newVendor.SapCode = vendor.SapCode
+
+	return v.vendorDBAccessor.Put(ctx, newVendor)
```

Comment:
I don't think the GetById is needed in this case, since in the accessor you update all the fields without needing the current vendor's data. You also don't need to validate if the current vendor exist, because the db itself will throw an error if it doesn't exist.

Reply:
Agreed